lib/season.rb

```diff
@@ -37,22 +37,29 @@ def self.generate_seasons(games)
     seasons_hash
   end
 
-  def shots_by_team_per_season_avg #helper method for most and least accurate
-    shots_by_teams = Hash.new(1)
+  def shots_by_team_per_season_avg(season_id)
+    shots_and_goals = Hash.new{|hash, team_name| hash[team_name] = Hash.new(0)}
 
     @games_in_season.each do |game|
-      shots_by_teams[game.teams_game_stats[:home_team][:team_name]] *= game.teams_game_stats[:home_team][:shots].to_f / game.teams_game_stats[:home_team][:goals]
-
-      shots_by_teams[game.teams_game_stats[:away_team][:team_name]] *= game.teams_game_stats[:away_team][:shots].to_f / game.teams_game_stats[:away_team][:goals]
+      away_team = game.teams_game_stats[:away_team][:team_name]
+      home_team = game.teams_game_stats[:home_team][:team_name]
+      shots_and_goals[away_team][:goals] += game.teams_game_stats[:away_team][:goals]
+      shots_and_goals[home_team][:goals] += game.teams_game_stats[:home_team][:goals]
+      shots_and_goals[home_team][:shots] += game.teams_game_stats[:home_team][:shots]
+      shots_and_goals[away_team][:shots] += game.teams_game_stats[:away_team][:shots]
+    end
+    team_avgs = {}
+    shots_and_goals.each do |team_name, stats|
+      team_avgs[team_name] = (stats[:goals].to_f / stats[:shots]).round(2)
```

Comment:
The only thing that I can see that might be throwing off the spec harness results is the rounding here. Maybe there a couple teams with very similar accuracies and rounding gives them the same accuracy. Since we're not displaying this accuracy there's no need to round it. Let me know if that works!